packages/api-sdk/src/medical/models/fhir.ts

```diff
@@ -94,3 +80,40 @@ export const resourceTypeForConsolidation = [
 ] as const;
 
 export type ResourceTypeForConsolidation = (typeof resourceTypeForConsolidation)[number];
+
+export const resourceSchema = z.array(z.enum(resourceTypeForConsolidation));
+
+export const consolidationConversionType = ["html", "pdf", "json"] as const;
+export type ConsolidationConversionType = (typeof consolidationConversionType)[number];
+
+export const consolidatedFilterSchema = z.object({
+  resources: z.string(),
+  dateFrom: z.date().optional(),
+  dateTo: z.date().optional(),
+});
+export type ConsolidatedFilter = z.infer<typeof consolidatedFilterSchema>;
+
+export const consolidatedCountSchema = z.object({
+  total: z.number(),
+  resources: z.record(z.number()),
+  filter: consolidatedFilterSchema,
+});
+
+export type ConsolidatedCountResponse = z.infer<typeof consolidatedCountSchema>;
+
+export const getConsolidatedFiltersSchema = z.object({
```

Comment:
Do we need both `consolidatedFilterSchema` and `getConsolidatedFiltersSchema`?

Reply:
extended `consolidatedFilterSchema`

Reply:
Yea the dates are different

Reply:
Ones string for filters we pass in and the other is dates

Reply:
NABD for this PR (we can address on a follow-up one):
- [bug] the SDK's `countPatientConsolidated()` returns `ConsolidatedCountResponse`, which has the dates as `Date`, but the endpoint returns those as `string`;
- [improvement] `getConsolidatedFiltersSchema` is extending `consolidatedFilterSchema`, but it overrides all properties - so no reason to extend it;
- [improvement] we probably can make `consolidatedCountSchema` use `getConsolidatedFiltersSchema` instead of `consolidatedFilterSchema`, it would simplify it
   - I'd look into using `Date` instead of `string`